bench.go

```diff
@@ -198,14 +206,21 @@ func (c *connectionBenchmark) runRateLimited() (time.Duration, error) {
 		}
 
 		before := time.Now()
-		if err := c.requester.Request(); err != nil {
-			return 0, err
-		}
+		err := c.requester.Request()
 		latency := time.Since(before).Nanoseconds()
+		if err != nil {
+			if err := c.errorHistogram.RecordCorrectedValue(latency, interval); err != nil {
+				return 0, err
+			}
+			if err := c.unErrorHistogram.RecordValue(latency); err != nil {
+				return 0, err
+			}
+			c.totalErrors++
+		}
```

Comment:
Are you intending to double count error latencies? Seems like they should only be included in the error histograms.